stl/inc/ranges

```diff
@@ -6837,6 +6837,602 @@ namespace ranges {
+            requires _Zip::_All_forward<_IsConst, _ViewTypes...> {
+            const auto _Temp = *this;
+            ++(*this);
+
+            return _Temp;
+        }
+
+        constexpr _Iterator& operator--() noexcept(_Is_nothrow_predecrementable) // strengthened
+            requires _Zip::_All_bidirectional<_IsConst, _ViewTypes...> {
+            _Zip::_Tuple_for_each([](auto& _Itr) { --_Itr; }, _M_Current);
+            return *this;
+        }
+
+        constexpr _Iterator operator--(int) noexcept(
+            _Is_nothrow_predecrementable&& is_nothrow_copy_assignable_v<_Iterator<_IsConst>>) // strengthened
+            requires _Zip::_All_bidirectional<_IsConst, _ViewTypes...> {
+            const auto _Temp = *this;
+            --(*this);
+
+            return _Temp;
+        }
+
+        constexpr _Iterator& operator+=(difference_type _X) noexcept(
+            (noexcept(_STD declval<iterator_t<_Maybe_const<_IsConst, _ViewTypes>>>() +=
+                      iter_difference_t<iterator_t<_Maybe_const<_IsConst, _ViewTypes>>>{_X})
+                && ...)) // strengthened
+            requires _Zip::_All_random_access<_IsConst, _ViewTypes...> {
+            _Zip::_Tuple_for_each(
+                [&_X]<typename _IteratorType>(_IteratorType& _Itr) { _Itr += iter_difference_t<_IteratorType>{_X}; },
+                _M_Current);
+
+            return *this;
+        }
+
+        constexpr _Iterator& operator-=(difference_type _X) noexcept(
+            (noexcept(_STD declval<iterator_t<_Maybe_const<_IsConst, _ViewTypes>>>() -=
+                      iter_difference_t<iterator_t<_Maybe_const<_IsConst, _ViewTypes>>>{_X})
+                && ...)) // strengthened
+            requires _Zip::_All_random_access<_IsConst, _ViewTypes...> {
+            _Zip::_Tuple_for_each(
+                [&_X]<typename _IteratorType>(_IteratorType& _Itr) { _Itr -= iter_difference_t<_IteratorType>{_X}; },
+                _M_Current);
+
+            return *this;
+        }
+
+        _NODISCARD constexpr auto operator[](difference_type _X) const
+            noexcept((noexcept((_STD declval<iterator_t<_Maybe_const<_IsConst, _ViewTypes>>>())
+                              [iter_difference_t<iterator_t<_Maybe_const<_IsConst, _ViewTypes>>>{_X}])
+                      && ...)) // strengthened
+            requires _Zip::_All_random_access<_IsConst, _ViewTypes...> {
+            return _Zip::_Tuple_transform(
+                [&_X]<typename _IteratorType>(
+                    _IteratorType& _Itr) -> decltype(auto) { return _Itr[iter_difference_t<_IteratorType>{_X}]; },
+                _M_Current);
+        }
+
+        _NODISCARD_FRIEND constexpr bool operator==(const _Iterator& _Lhs, const _Iterator& _Rhs) noexcept(
+            noexcept(_Lhs._M_Current == _Rhs._M_Current)) // strengthened
+            requires(equality_comparable<iterator_t<_Maybe_const<_IsConst, _ViewTypes>>>&&...) {

```

Comment:
```suggestion
        _NODISCARD_FRIEND constexpr bool operator==(const _Iterator& _Lhs, const _Iterator& _Rhs) noexcept(
            noexcept(_Implicitly_convert_to<bool>(_Lhs._M_Current == _Rhs._M_Current))) // strengthened
            requires(equality_comparable<iterator_t<_Maybe_const<_IsConst, _ViewTypes>>>&&...) {
```
We should check if conversion to `bool` throws too.

Reply:
If this isn't necessary, then I'll just mark the conversation as resolved without any necessary commits.